packages/cache/src/CacheStore.spec.tsx

```diff
@@ -32,98 +32,137 @@ describe('CacheStore', () => {
-    expect(cacheStore.getData(successCache(1))).toBeUndefined()
+  describe('reset', () => {
+    it('should reset data of all cached', async () => {
+      try {
+        cacheStore.suspend(successCache(1))
+      } catch (promiseToSuspense) {
+        await promiseToSuspense
+      }
+      try {
+        cacheStore.suspend(successCache(2))
+      } catch (promiseToSuspense) {
+        await promiseToSuspense
+      }
+      expect(cacheStore.getData(successCache(1))).toBe(TEXT)
+      expect(cacheStore.getData(successCache(2))).toBe(TEXT)
+      cacheStore.reset()
+      expect(cacheStore.getData(successCache(1))).toBeUndefined()
+      expect(cacheStore.getData(successCache(2))).toBeUndefined()
+    })
+
+    it('should reset data of specific cached', async () => {
+      try {
+        cacheStore.suspend(successCache(1))
+      } catch (promiseToSuspense) {
+        await promiseToSuspense
+      }
+      expect(cacheStore.getData(successCache(1))).toBe(TEXT)
+      cacheStore.reset(successCache(1))
+      expect(cacheStore.getData(successCache(1))).toBeUndefined()
+    })
   })
 
-  it("have getData method with key should get data of key's cached", async () => {
-    render(
-      <CacheStoreProvider store={cacheStore}>
-        <Suspense fallback={FALLBACK}>
-          <Cache {...errorCache(1)} cacheFn={() => sleep(ms('0.1s')).then(() => TEXT)}>
-            {(cached) => <>{cached.data}</>}
-          </Cache>
-        </Suspense>
-      </CacheStoreProvider>
-    )
+  describe('getData', () => {
+    it('should get data of specific cached', async () => {
+      render(
+        <CacheStoreProvider store={cacheStore}>
+          <Suspense fallback={FALLBACK}>
+            <Cache {...errorCache(1)} cacheFn={() => sleep(ms('0.1s')).then(() => TEXT)}>
+              {(cached) => <>{cached.data}</>}
+            </Cache>
+          </Suspense>
+        </CacheStoreProvider>
+      )
 
-    expect(screen.queryByText(FALLBACK)).toBeInTheDocument()
-    expect(screen.queryByText(TEXT)).not.toBeInTheDocument()
-    expect(cacheStore.getData(errorCache(1))).toBeUndefined()
-    await waitFor(() => expect(screen.queryByText(TEXT)).toBeInTheDocument())
-    expect(screen.queryByText(FALLBACK)).not.toBeInTheDocument()
-    expect(cacheStore.getData(errorCache(1))).toBe(TEXT)
+      expect(screen.queryByText(FALLBACK)).toBeInTheDocument()
+      expect(screen.queryByText(TEXT)).not.toBeInTheDocument()
+      expect(cacheStore.getData(errorCache(1))).toBeUndefined()
+      await waitFor(() => expect(screen.queryByText(TEXT)).toBeInTheDocument())
+      expect(screen.queryByText(FALLBACK)).not.toBeInTheDocument()
+      expect(cacheStore.getData(errorCache(1))).toBe(TEXT)
+    })
   })
 
-  it('should handle unsubscribe gracefully when no subscribers exist', () => {
-    const mockSync = vi.fn()
-    const cacheKey = ['nonexistent', 'key'] as const
-    cacheStore.unsubscribe({ cacheKey }, mockSync)
+  describe('unsubscribe', () => {
```

Comment:
How about below 2 things after this Pull Request
1. Remove unsubscribe method
2. CacheStore extends Subscribable https://github.com/TanStack/query/blob/main/packages/query-core/src/subscribable.ts